packages/process/src/node/process.ts

```diff
@@ -57,26 +55,10 @@ export abstract class Process implements Disposable {
 
     }
 
-    dispose() {
-        this.processManager.delete(this);
-
-        if (this.killed === false) {
-            const p = new Promise<void>(resolve => {
-                this.kill();
-                this.exitEmitter.event(event => resolve());
-            });
-            p.then(() => {
-                this.exitEmitter.dispose();
-                this.errorEmitter.dispose();
-            });
-        } else {
-            this.exitEmitter.dispose();
```

Comment:
Emitters don't seem to be disposed. Is not it required to remove references from listeners?

Reply:
As long as you clean the event handlers like:
in base-terminal-server.ts
```typescript
toDispose.push(term.onError(error => {
```
There's no more refs to the emitters and they do get cleaned-up see this diff between memory when a terminal is there and after I close it:
![dipose-emitters](https://user-images.githubusercontent.com/698076/30479958-791eafd2-99e5-11e7-8fb7-2adb7da91069.png)